src/sortable.js

```diff
@@ -23,116 +23,110 @@ angular.module('ui.sortable', [])
-            } else if (wrappers[key]) {
-              value = wrappers[key](value);
+            } else if ( key === 'helper' && typeof value === 'function') {
+              value = wrappersHelper(value);
             }
 
             // patch the options that need to have values set
             if (!value && (key === 'items' || key === 'ui-model-items')) {
               value = uiSortableConfig.items;
             }
-
             return value;
           }
 
           function patchUISortableOptions(newVal, oldVal, sortableWidgetInstance) {
-            function addDummyOptionKey(value, key) {
-              if (!(key in opts)) {
+            
+            // for this directive to work we have to attach some callbacks
+            angular.forEach(callbacks, function (value, key) {
+              if (!opts[key]) {
                 // add the key in the opts object so that
                 // the patch function detects and handles it
                 opts[key] = null;
               }
-            }
-            // for this directive to work we have to attach some callbacks
-            angular.forEach(callbacks, addDummyOptionKey);
-
-            // only initialize it in case we have to
-            // update some options of the sortable
-            var optsDiff = null;
-
-            if (oldVal) {
-              // reset deleted options to default
-              var defaultOptions;
-              angular.forEach(oldVal, function(oldValue, key) {
-                if (!newVal || !(key in newVal)) {
-                  if (key in directiveOpts) {
-                    if (key === 'ui-floating') {
-                      opts[key] = 'auto';
-                    } else {
-                      opts[key] = patchSortableOption(key, undefined);
-                    }
-                    return;
-                  }
-                  
-                  if (!defaultOptions) {
-                    defaultOptions = angular.element.ui.sortable().options;
-                  }
-                  var defaultValue = defaultOptions[key];
-                  defaultValue = patchSortableOption(key, defaultValue);
+            });
 
-                  if (!optsDiff) {
-                    optsDiff = {};
-                  }
-                  optsDiff[key] = defaultValue;
-                  opts[key] = defaultValue;
+            // initialize it in case we have to update some options of the sortable
+            var optsDiff = {};
+
+            // reset deleted options to default
+            var defaultOptions = angular.element.ui.sortable().options;
+            angular.forEach(oldVal, function(oldValue, key) {
+              if (!newVal || !newVal[key]) {
+                if (directiveOpts[key]) {
+                  opts[key] = key === 'ui-floating' ? 'auto' : patchSortableOption(key);
+                  return;
                 }
-              });
-            }
+                opts[key] = optsDiff[key] = patchSortableOption(key, defaultOptions[key]);
```

Comment:
There was some lazy laoding code here in order to avoid unnecessary object creations and (as a result) GC cycles.